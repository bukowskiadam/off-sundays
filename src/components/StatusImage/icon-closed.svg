<?xml version="1.0" encoding="utf-8"?>
<svg viewBox="0 0 500 500" xmlns="http://www.w3.org/2000/svg" xmlns:bx="https://boxy-svg.com">
  <defs>
    <bx:grid x="0" y="0" width="5.487" height="5.487"/>
    <linearGradient id="gradient-3" gradientUnits="userSpaceOnUse" x1="250" y1="50.001" x2="250" y2="449.999" gradientTransform="matrix(0.743378, -0.485173, 0.637648, 0.798288, -75.086223, 173.778856)">
      <stop offset="0" style="stop-color: rgba(255, 49, 49, 1)"/>
      <stop offset="1" style="stop-color: rgba(202, 0, 0, 1)"/>
    </linearGradient>
    <linearGradient id="gradient-0" gradientUnits="userSpaceOnUse" x1="461.769" y1="71.612" x2="461.769" y2="521.612" gradientTransform="matrix(0.867317, -0.244033, 0.428617, 1.210588, -80.840869, 101.12507)">
      <stop offset="0" style="stop-color: rgb(248, 248, 248);"/>
      <stop offset="1" style="stop-color: rgb(201, 201, 201);"/>
    </linearGradient>
  </defs>
  <g transform="matrix(1.211013, 0, 0, 1.211013, -62.221859, -55.117451)">
    <ellipse transform="matrix(1, 0.000023, -0.000023, 1, -105.204339, -48.439096)" cx="363.03" cy="300.383" rx="202.723" ry="202.723" style="fill: url(#gradient-0);" bx:origin="0.523 0.499"/>
    <polyline style="fill: none; stroke-linecap: round; stroke-linejoin: round; stroke: rgb(49, 49, 49); stroke-width: 30;" points="156.121 191.711 174.41 191.711 219.301 307.025 331.987 306.953 364.869 232.882 197.208 232.828" transform="" bx:origin="0.551 0.516"/>
    <path d="M 132.299 377.471 L 383.337 126.433 M 435.356 251.952 C 435.356 350.004 355.87 429.49 257.816 429.49 C 159.766 429.49 80.281 350.004 80.281 251.95 C 80.281 153.9 159.767 74.415 257.818 74.415 C 355.87 74.415 435.356 153.901 435.356 251.952 Z" style="stroke-width: 50; fill: none; stroke: url(#gradient-3); stroke-opacity: 0.78;" bx:origin="0.526 0.498"/>
    <ellipse cx="235.438" cy="350.422" rx="19.999" ry="19.999" style="fill: rgb(49, 49, 49);" bx:origin="1.542 -1.966"/>
    <ellipse transform="matrix(1, -0.000003, 0.000005, 1, 117.065447, 34.277308)" cx="198.289" cy="316.145" rx="19.999" ry="19.999" style="fill: rgb(49, 49, 49);" bx:origin="-0.698 -1.966"/>
  </g>
</svg>