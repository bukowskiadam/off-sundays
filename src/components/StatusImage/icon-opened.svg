<?xml version="1.0" encoding="utf-8"?>
<svg viewBox="0 0 500 500" xmlns="http://www.w3.org/2000/svg" xmlns:bx="https://boxy-svg.com">
  <defs>
    <bx:grid x="0" y="0" width="5.487" height="5.487"/>
    <pattern id="pattern-0" x="0" y="0" width="25" height="25" patternUnits="userSpaceOnUse" viewBox="0 0 100 100">
      <rect x="0" y="0" width="50" height="100" style="fill: black;"/>
    </pattern>
    <linearGradient id="gradient-2" gradientUnits="userSpaceOnUse" x1="257.819" y1="49.642" x2="257.819" y2="454.262" gradientTransform="matrix(0.88705, -0.461673, 0.515911, 1.020409, -98.188202, 113.886422)">
      <stop offset="0" style="stop-color: rgb(109, 139, 214);"/>
      <stop offset="1" style="stop-color: rgb(18, 32, 181);"/>
    </linearGradient>
  </defs>
  <g transform="matrix(1.112155, 0, 0, 1.112155, -36.734661, -30.20966)">
    <rect x="55.509" y="49.642" width="404.62" height="404.62" rx="24.598" ry="24.598" style="fill: url(#gradient-2);"/>
    <rect x="73.492" y="67.625" width="368.654" height="368.654" rx="7.678" ry="7.678" style="fill: none; stroke: rgb(255, 255, 255); opacity: 0.95; stroke-width: 14.3865;"/>
  </g>
  <g transform="matrix(1.346121, 0, 0, 1.346121, -107.657776, -111.348831)">
    <polyline style="fill: none; stroke-linecap: round; stroke-linejoin: round; stroke-width: 30; stroke: rgb(255, 255, 255); opacity: 0.95;" points="156.121 191.711 174.41 191.711 219.301 307.025 331.987 306.953 364.869 232.882 197.208 232.828" bx:origin="0.551 0.516"/>
    <ellipse cx="235.438" cy="350.422" rx="19.999" ry="19.999" style="fill: rgb(255, 255, 255); opacity: 0.95;" bx:origin="1.542 -1.966"/>
    <ellipse transform="matrix(0.999999, -0.000003, 0.000005, 1, 117.065721, 34.277308)" cx="198.289" cy="316.145" rx="19.999" ry="19.999" style="fill: rgb(255, 255, 255); opacity: 0.95;" bx:origin="-0.698 -1.966"/>
  </g>
</svg>