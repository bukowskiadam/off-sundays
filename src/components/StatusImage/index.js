import React from 'react';
import PropTypes from 'prop-types';

import imageClosed from './icon-closed.svg';
import imageOpened from './icon-opened.svg';

const StatusImage = ({ isClosed }) => {
    const image = isClosed ? imageClosed : imageOpened;
    const altText = isClosed ? 'zamkniÄ™te' : 'otwarte';

    return <img src={image} alt={altText} />;
};

StatusImage.propTypes = {
    isClosed: PropTypes.bool.isRequired,
};

export default StatusImage;
