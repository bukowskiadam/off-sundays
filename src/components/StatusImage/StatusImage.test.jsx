import React from 'react';
import { shallow } from 'enzyme';

import StatusImage from './StatusImage';

describe('StatusImage', () => {
    it('renders correctly for closed', () => {
        const wrapper = shallow(<StatusImage isClosed={true} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly for opened', () => {
        const wrapper = shallow(<StatusImage isClosed={false} />);

        expect(wrapper).toMatchSnapshot();
    });
});
