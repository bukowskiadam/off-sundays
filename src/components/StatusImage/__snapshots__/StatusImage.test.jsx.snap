// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusImage renders correctly for closed 1`] = `
<img
  alt="zamkniÄ™te"
  src="icon-closed.svg"
/>
`;

exports[`StatusImage renders correctly for opened 1`] = `
<img
  alt="otwarte"
  src="icon-opened.svg"
/>
`;
